src/libmeasurement_kit/ooni/resources_impl.hpp

```diff
@@ -0,0 +1,200 @@
+
+template <MK_MOCK_NAMESPACE(http, get)>
+void get_manifest_as_json_impl(
+        std::string latest, Callback<Error, nlohmann::json> callback,
+        Settings settings, Var<Reactor> reactor, Var<Logger> logger) {
+    std::string url{
+      "https://github.com/OpenObservatory/ooni-resources/releases/download/"
+    };
+    url += latest;
+    url += "/manifest.json";
+    if (settings.find("http/max_redirects") == settings.end()) {
+        settings["http/max_redirects"] = 4;
+    }
+    logger->info("Downloading manifest; please, be patient...");
+    http_get(url, [=](Error error, Var<Response> response) {
+        nlohmann::json result;
+        if (error) {
+            callback(error, result);
+            return;
+        }
+        if (response->status_code != 200) {
+            callback(CannotGetResourcesManifestError(), result);
+            return;
+        }
+        try {
+            result = nlohmann::json::parse(response->body);
+        } catch (const std::invalid_argument &) {
+            callback(JsonParseError(), result);
+            return;
+        }
+        logger->info("Downloaded manifest");
+        callback(NoError(), result);
+    }, {}, settings, reactor, logger, nullptr, 0);
+}
+
+static inline bool ostream_bad(const std::ostream &s) {
+    return s.bad();
+}
+
+template <MK_MOCK_NAMESPACE(http, get), MK_MOCK(ostream_bad)>
+void get_resources_for_country_impl(std::string latest, nlohmann::json manifest,
+                                    std::string country, Callback<Error> cb,
+                                    Settings settings, Var<Reactor> reactor,
+                                    Var<Logger> logger) {
+    if (!manifest.is_object()) {
+        reactor->call_soon([=]() { cb(JsonDomainError()); });
+        return;
+    }
+    if (manifest.find("resources") == manifest.end()) {
+        reactor->call_soon([=]() { cb(JsonKeyError()); });
+        return;
+    }
+    if (settings.find("http/max_redirects") == settings.end()) {
+        settings["http/max_redirects"] = 4;
+    }
+    // TODO: do we need to cross validate latest?
+    std::vector<Continuation<Error>> input;
+    /*
+     * Important: MUST be entry and not &entry because we need a copy!
+     */
+    for (auto entry : manifest["resources"]) {
+        input.push_back([=](Callback<Error> cb) {
+            if (!entry.is_object()) {
+                cb(JsonDomainError());
+                return;
+            }
+            if (entry.find("country_code") == entry.end()) {
+                cb(JsonKeyError());
+                return;
+            }
+            std::string cc = entry["country_code"];
+            if (country != cc and country != "ALL") {
+                cb(NoError()); // Nothing to be done for this entry
+                return;
+            }
+            if (entry.find("path") == entry.end()) {
+                cb(JsonKeyError());
+                return;
+            }
+            std::string path = entry["path"];
```

Comment:
> You probably want to do some checks to ensure that entry["path"] doesn't contain a bad filepath. If ooni-resources tells you path is "/etc/passsd" you will end up writing there.

AFAICT, path was already scrubbed from `/`. I'll add support for scrubbing `\\` as well.

> It would also be better if this function supported passing a dst path as a settings or as an argument.

Okay, I will implement that.